Pages/Shipment/ShipmentRegistration.razor

```diff
@@ -1,84 +1,81 @@
-                    {
-                        <option value="@machine.Id">@machine.Address</option>
-                    }
+                    <option value="@machine.Id">@machine.Address</option>
                 }
+            }
         </select><br/>
 
-        <label>Destination machine: </label>
+        <label>Destination machine: </label><br/>
         <select @bind="Shipment.DestinationMachineId">
-                @if(PostMachines != null)
+            @if (PostMachines != null)
+            {
+                @foreach (var machine in PostMachines)
                 {
-                    @foreach(var machine in PostMachines)
-                    {
-                        <option value="@machine.Id">@machine.Address</option>
-                    }
+                    <option value="@machine.Id">@machine.Address</option>
                 }
-        </select>
+            }
+        </select><br/>
 
-        
+        <button type="submit">Register shipment</button>
     </div>
-    <button type="submit">Register shipment</button>
+
 
 </EditForm>
 
 
 @code {
-    [Inject]
-    public AuthenticationStateProvider authStateProvider { get; set; }
+
+    [CascadingParameter]
+    Task<AuthenticationState> authStateTask { get; set; }
 
     [Inject]
-    public IShipmentRepository ShipmentRepository { get; set; }
+    ISenderRepository SenderRepository { get; set; }
 
-    public IPostMachineRepository PostMachineRepository => Service;
+    [Inject]
+    IShipmentRepository ShipmentRepository { get; set; }
 
-    public IEnumerable<PostMachine> PostMachines { get; set; }
-    public Shipment Shipment { get; set; }
+    IPostMachineRepository PostMachineRepository => Service;
+    Sender sender;
+    IEnumerable<PostMachine> PostMachines { get; set; }
+    Shipment Shipment { get; set; }
 
     protected override async Task OnInitializedAsync()
     {
-        Shipment = new Shipment();
-        
-        //This should get id of logged in user and use it to set senderId of shipment.
-        //Can't be used yet as log in is not yet implemented.
-
-        //var user = (await authStateProvider.GetAuthenticationStateAsync()).User;
-        //Shipment.SenderId = user.FindFirst(c => c.Type == "sub")?.Value;
-        
-
+        var authState = await authStateTask;
+        var username = authState.User.Identities.First().Name;
+        sender = SenderRepository.Senders.First(s => s.Username == username);
+        Shipment = new Shipment { SenderId = sender.Id };
         RetrievePostMachines();
     }
 
-    public void RetrievePostMachines()
+    void RetrievePostMachines()
     {
         PostMachines = PostMachineRepository.PostMachines;
     }
 
-    public async void CreateShipment()
+    async void HandleSubmit()
     {
         await ShipmentRepository.CreateAsync(Shipment);
-    }
-
-    public async void HandleSubmit()
-    {
-        CreateShipment();
+        Shipment = new Shipment { SenderId = sender.Id };
```

Comment:
For some reason this does not clear the fields. Maybe browser autofills them?

Reply:
Outdated. Resolving, then, I guess.